TACC/Core-Portal
# task/FP-545: Create _common Button React Component

## Overview: ##
Made _common Button component.

## Related Jira tickets: ##

* [FP-545](https://jira.tacc.utexas.edu/browse/FP-545)
* [FP-1312](https://jira.tacc.utexas.edu/browse/FP-1312)

## Summary of Changes: ##

- Added a Button component to _common which is built from HTML `<button>`
- Defined Button.module.css which imports styles from styles/components/c-button.css
- Added UIPatterns button, which shows all of the possible configurations of the _common Button

## Testing Steps: ##
1. Navigate to UI Patterns, and scroll to the bottom
2. Confirm that rendered Button patterns address the styles given [here](https://confluence.tacc.utexas.edu/display/UP/UI+-+Components+-+Button) and [here](https://xd.adobe.com/view/bdd30952-4f4a-4e8d-5a42-29670859679c-cbe2/)
3. *Refer to [this design thread on Slack](https://tacc-team.slack.com/archives/CQUA4D5KJ/p1641930563005500?thread_ts=1641928526.004400&cid=CQUA4D5KJ) for more styles that may not be addressed in either document above.

## UI Photos:

https://user-images.githubusercontent.com/43251554/153292381-307f9ddd-bbf1-4a1a-96df-424252fdef5d.mov

## Notes: ##
Please note more styles to add for this component that I may have overlooked!
** Just realized I need to write up a test script, expect that in a future commit **
